### PR TITLE
#13413: Fix all_gather sweep test

### LINKED ISSUE #13413: [Bug Report] CCL sweep test is in bad state
**Describe the bug**
Mesh setup is not properly setup for the sweep test

Screenshot 2024-10-03 at 8 29 40 PM

### PR DESCRIPTION
**Pull request by Aswinmcw**
Tracking Issue : #13413 

Fix mesh setup that causes issue while running

### Checklist
- [x] Post commit CI passes - https://github.com/tenstorrent/tt-metal/actions/runs/11159556556
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by SeanNijjar** [comment]
Hey @cfjchu - does this change make sense for line all-gather? I'm not familiar with the details of this API if we are going past 4 chips of a t3k. I suspect what we need to do here is wrap or reuse `pcie_mesh_device` and `t3k_mesh_device`in this sweep like we did for test_all_gather

**Comment by Aswinmcw** [comment]
Screenshot 2024-10-03 at 8 29 00 PM
This is the error i am getting

**Comment by Aswinmcw** [comment]
> Screenshot 2024-10-03 at 8 29 00 PM This is the error i am getting

If I am correct the mesh device api is taking the params as kw args, as we passed the values alone, it assumes the device ids as the 2nd param `l1_small_size` which should not happen. @SeanNijjar @cfjchu ?

**Comment by SeanNijjar** [comment]
This looks different from the t3k device mesh
```
    mesh_device = ttnn.open_mesh_device(
        ttnn.MeshShape(2, 4),
        dispatch_core_type=get_dispatch_core_type(),
        **device_params,
        mesh_type=ttnn.MeshType.Ring,
    )
 ```

Taken from conftest.py

I'm guessing (hoping) that it's enough to just change the `mesh_type` to `ttnn.MeshType.Linear` but I am not sure. To be honest, we reuse the t3k_mesh_device for line all gather tests so it should "just work" without additional changes. Is it possible to import this directly from the conftest?

**Comment by Aswinmcw** [comment]
> This looks different from the t3k device mesh
> 
> ```
>     mesh_device = ttnn.open_mesh_device(
>         ttnn.MeshShape(2, 4),
>         dispatch_core_type=get_dispatch_core_type(),
>         **device_params,
>         mesh_type=ttnn.MeshType.Ring,
>     )
> ```
> 
> Taken from conftest.py
> 
> I'm guessing (hoping) that it's enough to just change the `mesh_type` to `ttnn.MeshType.Linear` but I am not sure. To be honest, we reuse the t3k_mesh_device for line all gather tests so it should "just work" without additional changes to mesh_type. Is it possible to import this directly from the conftest?

I guess for sweeps we can't use the fixtures from conftest file, that is the reason why they introduced a device generator within the sweep test itself

File: tests/sweep_framework/sweeps/ccl/line_all_gather.py [hunk 1]
```diff
@@ -65,7 +65,9 @@ def mesh_device_fixture():
     assert ttnn.get_num_devices() >= 8, "Not T3000!"
     device_ids = ttnn.get_t3k_physical_device_ids_ring()
     num_devices_requested = len(device_ids)
-    mesh_device = ttnn.open_mesh_device(ttnn.MeshShape(1, num_devices_requested), device_ids[:num_devices_requested])
+    mesh_device = ttnn.open_mesh_device(
+        ttnn.MeshShape(1, num_devices_requested), physical_device_ids=device_ids[:num_devices_requested]
```
Comment (cfjchu) [inline tests/sweep_framework/sweeps/ccl/line_all_gather.py:69]: This does not guarantee that the devices are connected in a line (if that's what you're testing). Please do this instead:

`mesh_device = ttnn.open_mesh_device(ttnn.MeshShape(1, num_devices_requested), mesh_type=ttnn.MeshType.Line)`

Comment (Aswinmcw) [reply tests/sweep_framework/sweeps/ccl/line_all_gather.py]: Updated

**Review by SeanNijjar — APPROVED** [review]

**Review by jdesousa-TT — APPROVED** [review]

File: tests/sweep_framework/sweeps/ccl/line_all_gather.py (hunk 1 above)
Comment (cfjchu) [reply tests/sweep_framework/sweeps/ccl/line_all_gather.py]: excellent, thanks!

**Pull request merged**